File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/chain_of_responsibility/solution/CustomerServiceManager.java
// CustomerServiceManager.java

public class CustomerServiceManager extends Handler
{
	public CustomerServiceManager()
	{
		super(null);
	}

	protected boolean canHandle(CustomerRequestsDiscountEvent aCustomerRequestsDiscountEvent)
	{
		return (aCustomerRequestsDiscountEvent.getPercent() < 30);
	}

	protected void doHandle(CustomerRequestsDiscountEvent aCustomerRequestsDiscountEvent)
	{
		System.out.println("I am a CustomerServiceManager.");
		System.out.println("I can give you a " + aCustomerRequestsDiscountEvent.getPercent() + "% discount.");
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/chain_of_responsibility/solution/CustomerServiceManager.m
% CustomerServiceManager.m

classdef CustomerServiceManager < Handler
	methods
		% constructor:
		function theCustomerServiceManager = CustomerServiceManager
			theCustomerServiceManager@Handler({});
		end
	end

	methods (Access = protected)
		function result = canHandle(theCustomerServiceManager, aCustomerRequestsDiscountEvent)
			result = (aCustomerRequestsDiscountEvent.Percent < 30);
		end

		function doHandle(theCustomerServiceManager, aCustomerRequestsDiscountEvent)
			fprintf('I am a CustomerServiceManager.\n');
			fprintf('I can give you a %d%% discount.\n', aCustomerRequestsDiscountEvent.Percent);
		end
	end
end
